### DISCUSSION TITLE
Feature ideas?

### DISCUSSION THREAD

**Discussion by grinapo** [ClementTsang/bottom#853] [General]
I'm not sure how do you prefer wishlist items (if at all). I'll just met btm and would like to throw in some.

- [processes] io rate; right now there is read and write rate but no **sum** of those, and it would be really useful for sorting (keep active processes up)
- [processes] hide all zero stat/sleeping/unknown process (or even for Disks, less useful though)
- [disks] sort disks by activity (read, write, both), used/free (https://github.com/ClementTsang/bottom/issues/567)
- [disks] used disk in GB (instead of %)
- [network] it ought to show somehow interfaces; config suggests I can filter them, but cannot see them :-)
- [help] please do not quit by Q; treat it as ESC. I always exit btm when want to close the help. Every. Damned. Time. ;-)
- [config] what is "show scroll pos" color? doesn't seem to do anything for me.
- [new] disk io graph widget
- [new][colors] there are some very useful visual ideas in htop, like coloring basedir different from basename different from args in full command line, or all zero fields gray and non-zero is colored, and even gradient coloring like 0% - 25% - 50% - + ). It's nice.

**Comment by grinapo** [comment]
Seems to be connected to https://github.com/ClementTsang/bottom/issues/374 in multiple places. :-)

**Comment by ClementTsang** [comment]
Thanks for the list of suggestions, I'll definitely consider them and I can make new issues for some later. Just some comments on some of these for now:

- Sum of IO: like, total inputs/outputs for a process? As that's already there (total read/total write columns) - not sure what you mean otherwise.
- Used disk in GB: yep, was going to add that in a bit.
- Network interfaces: yep, this is something I want to address, but it might require some effort to make configurations editable within the application, which is a bit of a rabbit hole for me.
- Q: Can look into that, was going to make shortcuts more configurable in general too.
- scroll: not sure what you're talking about, could you show a screenshot?

**Reply by grinapo** [reply]
>     * Sum of IO: like, total inputs/outputs for a process? As that's already there (total read/total write columns) - not sure what you mean otherwise.

How do you sort the list that the most IO active process is at the top? :-)

>     * Network interfaces: yep, this is something I want to address, but it might require some effort to make configurations editable within the application, which is a bit of a rabbit hole for me.

I have noticed that you're in the middle of refactoring the widgets, so that's for the future.

>     * Q: Can look into that, was going to make shortcuts more configurable in general too.

Yep you seem to have openissues for that, too. 

>     * scroll: not sure what you're talking about, could you show a screenshot?

`show_table_scroll_position = true` which does not do anything for me, possibly because I have overridden the colors?

**Reply by ClementTsang** [reply]
> How do you sort the list that the most IO active process is at the top? :-)

Click the header, or use `s` to open the sort menu and select that process. This is documented [here](https://clementtsang.github.io/bottom/0.6.8/usage/widgets/process/#sorting), the sort menu is also in the help menu.

![image](https://user-images.githubusercontent.com/34804052/200159943-c98aa949-acd7-415a-8477-a1c23a431d0f.png)

Another reminder that I need to improve discoverability though within the application...

> `show_table_scroll_position`

That just shows the current scroll location for a table:
![image](https://user-images.githubusercontent.com/34804052/200159829-ccfb18f9-586c-4a97-a795-c12e68be3953.png)

This indicator is just themed the same as the rest of the widget title.

**Reply by grinapo** [reply]
> > How do you sort the list that the most IO active process is at the top? :-)
> 
> Click the header, or use `s` to open the sort menu and select that process. This is documented [here](https://clementtsang.github.io/bottom/0.6.8/usage/widgets/process/#sorting), the sort menu is also in the help menu.

:grin:  yes and which one do you choose?

I have _read io_ and _write io_, but I don't see anything which would sort by _io activity_ (=read+write).

**Reply by ClementTsang** [reply]
Ah, I see, didn't understand that you meant sum of read + write, your wording made me think just sum of read or sum of write. NGL, I don't really have much interest/time on hand to adding that right now (it wouldn't be hard) but I might look at it later when I add configurable columns.

**Comment by grinapo** [comment]
No need to hurry, write it at the end of your todo list ;-)
Thanks!
